src/EventListener/Pimcore/ChangeListener.php

```diff
@@ -48,7 +48,12 @@ public function handle(AssetEvent|DataObjectEvent|DocumentEvent $event): void
             return;
         }
 
-        $this->messageBus->dispatch(new RefreshElement($this->getFreshElement($element)));
+        try {
+            $this->messageBus->dispatch(new RefreshElement($this->getFreshElement($element)));
+        } catch (PimcoreElementNotFoundException) {
```

Comment:
Instead of relying on the try-catch, I'd rather introduce a new method to handle deletions and refactor common elements into a private method.